demos/portal-embed-demo/public/js/ttt/audio-controller.js

```diff
@@ -0,0 +1,369 @@
+                </div>
+                <div class='controller'>
+                    <img class='play' src='/img/play.png'>
+                </div>
+                <div class='controller'>
+                    <img class='next' src='/img/next.png'>
+                </div>
+            </div>
+        `
+    }
+
+    /**
+     * Controlling play button
+     */
+    play() {
+        this.playButton.src = '/img/pause.png';
+        const startAxis = _ => {
+            // Start the progress bar
+            const left = this.timeline.offsetWidth;
+            this.axis.style.transition = `${this.durationSec}s linear`;
+            this.axis.style.left = `${left}px`;
+            this.isPlaying = true;
+        }
+        // https://developers.google.com/web/updates/2017/06/play-request-was-interrupted
+        const promise = this.audio.play();
+        if(promise !== undefined) {
+            promise.then(_ => {
+                startAxis();
+            }).catch(e => {
+                console.log(e);
+            });
+            return;
+        }
+        startAxis();
+    }
+
+    /**
+     * Controlling pause button
+     */
+    pause() {
+        if(!this._isPlaying()) {
+            return;
+        }
+        this.playButton.src = '/img/play.png';
+        this.isPlaying = false;
+        this.audio.pause();
+        // Stop the progress bar
+        this.axis.style.left = `${this._getAxisCurrentLeft()}px`;
+    }
+
+    /**
+     * Controlling next button
+     */
+    next() {
+        if(!this._hasNext()){
+            return;
+        }
+        this.playlistIndex++;
+        this.prevButton.style.opacity = 1.0;
+        this._createNewAudio(this.playlist[this.playlistIndex]);
+        if(this._isPlaying()) {
+            // If the controller was playing an audio, 
+            // start playing the new audio
+            this.play();
+        }
+        this._updateController();
+    }
+
+    /**
+     * Controlling prev button
+     */
+    prev() {
+        if(!this._hasPrev()){
+            return;
+        }
+        this.playlistIndex--;
+        this.prevButton.style.opacity = 1.0;
+        this._createNewAudio(this.playlist[this.playlistIndex]);
+        if(this._isPlaying()) {
+            // If the controller was playing an audio, 
+            // start playing the new audio
+            this.play();
+        }
+        this._updateController();
+    }
+
+    /**
+     * Add audio to playlist
+     * @param {String} id - audio ID
+     */
+    addPlaylist(id) {
+        this.playlist.push(id);
+        this._updateController();
+    }
+
+    /**
+     * Remove audio from playlist
+     * @param {String} id - audio ID
+     */
+    removePlaylist(id) {
+        // Simply create new playlist filtering the ID to remove
+        const newPlaylist = [];
+        this.playlist
+            .filter(audioId => audioId !== id)
+            .map(audioId => newPlaylist.push(audioId));
```

Comment:
`filter` already produces the array you want:

```javascript
this.playlist = this.playlist.filter(audioId => audioId !== id);
```
`splice` would be more efficient, yet, assuming the ID can only occur once, but maybe that's not crucial for this demo:

```javascript
const idx = this.playlist.indexOf(audioId);
if (idx !== -1) this.playlist.splice(idx, 1);
```

Reply:
Ahhhh, please let me use your recommendation.
``` javascript
        // Remove given id from playlist
        const index = this.playlist.indexOf(audioId);
        if (index !== -1) this.playlist.splice(index, 1);
```